gui/rendering/Node.hpp

```diff
@@ -150,17 +162,51 @@ class Node : public std::enable_shared_from_this<Node> {
   virtual void setScale(QVector3D scale);
 
   /**
-   * Sets the m_dirty flag for the node so transforms are updated later.
+   *
+   * @return RenderPriority
+   */
+  RenderPriority getPriority() const;
+
+protected:
+  std::multiset<std::weak_ptr<rendering::Node>, CompareNodes> renderingQueue{};
```

Comment:
As this multiset is never updated after it is populated a more efficient data structure to use would be a std::vector that is sorted once after populating - it'll be faster to iterate over than the multiset